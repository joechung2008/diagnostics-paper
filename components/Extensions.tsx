import React, { useMemo } from "react";
import { ScrollView } from "react-native";
import { Button } from "react-native-paper";
import { useExtensionsStyles } from "../hooks/useExtensionsStyles";
import type { ExtensionsProps } from "../lib/types";
import { byKey, isExtensionInfo, toNavLink } from "../lib/utils";

const Extensions = ({ extensions, onLinkClick }: ExtensionsProps) => {
  const dynamicStyles = useExtensionsStyles();

  const links = useMemo(
    () =>
      Object.values(extensions)
        .filter(isExtensionInfo)
        .map(toNavLink)
        .sort(byKey),
    [extensions]
  );

  return (
    <ScrollView style={dynamicStyles.container} accessibilityLabel="Extensions">
      {links.map((link) => (
        <Button
          key={link.key}
          mode="outlined"
          onPress={() => onLinkClick?.(link)}
          style={dynamicStyles.button}
        >
          {link.name}
        </Button>
      ))}
    </ScrollView>
  );
};

export default Extensions;
