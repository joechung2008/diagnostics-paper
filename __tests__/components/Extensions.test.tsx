import React from "react";
import { PaperProvider, MD3LightTheme } from "react-native-paper";
import { render } from "@testing-library/react-native";
import Extensions from "../../components/Extensions";

describe("Extensions", () => {
  const mockOnLinkClick = jest.fn();

  it("renders correctly with extensions", () => {
    const mockExtensions = {
      ext1: {
        extensionName: "Extension One",
        manageSdpEnabled: true,
      },
      ext2: {
        extensionName: "Extension Two",
        manageSdpEnabled: false,
      },
    };

    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <Extensions extensions={mockExtensions} onLinkClick={mockOnLinkClick} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with empty extensions", () => {
    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <Extensions extensions={{}} onLinkClick={mockOnLinkClick} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
