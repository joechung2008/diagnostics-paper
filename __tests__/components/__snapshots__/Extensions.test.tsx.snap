// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extensions renders correctly with empty extensions 1`] = `
<RCTScrollView
  accessibilityLabel="Extensions"
  style={
    {
      "backgroundColor": "#FFFBFE",
      "flexGrow": 0,
      "flexShrink": 1,
      "minWidth": 0,
      "padding": 10,
    }
  }
>
  <View />
</RCTScrollView>
`;

exports[`Extensions renders correctly with extensions 1`] = `
<RCTScrollView
  accessibilityLabel="Extensions"
  style={
    {
      "backgroundColor": "#FFFBFE",
      "flexGrow": 0,
      "flexShrink": 1,
      "minWidth": 0,
      "padding": 10,
    }
  }
>
  <View>
    Button: Extension One
    Button: Extension Two
  </View>
</RCTScrollView>
`;
